src/Listeners/DeleteBreadPermission.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace TCG\Voyager\Listeners;
+
+use TCG\Voyager\Events\BreadDeleted;
+use TCG\Voyager\Facades\Voyager;
+use TCG\Voyager\Models\Permission;
+use TCG\Voyager\Models\Role;
+
+class DeleteBreadPermission
+{
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Delete Permission for a given BREAD.
+     *
+     * @param BreadDeleted $bread
+     *
+     * @return void
+     */
+    public function handle(BreadDeleted $bread)
+    {
+        if (config('voyager.bread.add_permission') && file_exists(base_path('routes/web.php'))) {
+            // Permission::generateFor(snake_case($bread->dataType->slug));
+            $role = Role::where('name', config('voyager.bread.default_role'))->firstOrFail();
```

Comment:
None of this is necessary.  I'd say it's safe to assume the database is set up correctly with ONDELETE, and will delete the associated pivot table rows when the permissions themselves are deleted.